pkg/scheduler/cache/event_handlers.go

```diff
@@ -258,6 +258,8 @@ func (sc *SchedulerCache) syncTask(oldTask *schedulingapi.TaskInfo) error {
 	newPod, err := sc.kubeClient.CoreV1().Pods(oldTask.Namespace).Get(context.TODO(), oldTask.Name, metav1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
+			sc.Mutex.Lock()
```

Comment:
I'm not sure if we need to use lock in notfound error as well.  this error is not common, but it works if we add it. 
nice catch! :)